docs/apache-airflow/security/access-control.rst

```diff
@@ -37,101 +37,113 @@ regarding its security model.
-using the ``airflow roles create`` command, e.g.:
+ .. image:: /img/list-roles.png
 
-.. code-block:: bash
+2. Specify a name and add permissions for the role. Note that the names for permissions will vary slightly between the UI and the source code.
 
-  airflow roles create Role1 Role2
+ .. image:: /img/add-permissions.png
 
-And we could assign the given role to a new user using the ``airflow
-users add-role`` CLI command.
+3. Click Save.
 
+You can also create roles via the CLI using the following command:
 
-Permissions
-'''''''''''
+.. code-block:: bash
+
+  airflow roles create <Role1> <Role2>
+
+You could then assign the given role to a new user using the ``airflow
+users add-role`` CLI command. Note that adding and removing permissions for a role can only be completed in the UI.
 
 Resource-Based permissions
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Starting with version 2.0, permissions are based on individual resources and a small subset of actions on those
+Starting with version 2.0, permissions are based on individual resources and a small set of actions with those
 resources. Resources match standard Airflow concepts, such as ``Dag``, ``DagRun``, ``Task``, and
 ``Connection``. Actions include ``can_create``, ``can_read``, ``can_edit``, and ``can_delete``.
 
-Permissions (each consistent of a resource + action pair) are then added to roles.
+Each endpoint in Airflow requires one or more permissions to be accessed. To access an endpoint, a role needs all permissions required by that endpoint.
 
-**To access an endpoint, the user needs all permissions assigned to that endpoint**
+DAG-Based Permissions
+^^^^^^^^^^^^^^^^^^^^^
+``Admin`` users can create roles with permission to access only specific DAGs. This is called DAG level access.
 
-There are five default roles: Public, Viewer, User, Op, and Admin. Each one has the permissions of the preceding role, as well as additional permissions.
+While access to all DAGs is controlled with resource-based permissions, DAG level access only has two associated permissions:
+``can_read`` and ``can_edit`` (``can_dag_read`` and ``can_dag_edit`` were deprecated in 2.0.0).
 
-DAG-level permissions
-^^^^^^^^^^^^^^^^^^^^^
+For instance, the following images show how you would create a role which can only write to
+``example_python_operator``. Once you create the role, the permission appears in the **List Roles** menu.
+
+.. image:: /img/add-role.png
+.. image:: /img/new-role.png
+
+If an endpoint for a specific DAG requires a permission based on the DAG resource (for example, ``DAGs.can_create``),
+a role will have access if it has either the resource-based permission or a DAG-based permission.
+
+For example, if an endpoint for the ``example_python_operator`` DAG requires the resource-based ``DAGS.can_read`` permission,
+a role with the DAG-based permission ``DAG:example_python_operator.can_read`` can still access that endpoint even without ``DAGS.can_read``.
```

Comment:
Do DAG-level permissions give you access to all resource-level actions for a DAG? Wondering if I should use an example with "can_read"/ "can_edit" or "can_create" to show the DAG-level permissions enable more than just the single action their name represents.

Reply:
So I think there's a bit of confusion around DAG permissions. 

* Any permission with the `DAGs` prefix, such as `DAGs.can_read`, `DAG.can_delete`, etc., applies to all DAGs.
* Any permission that starts with `DAG:<dag_name>`, such as `DAG:example_python_operator.can_read` or `DAG: DAG:example_python_operator.can_edit`, only applies to that specific DAG.

Unfortunately, there isn't a consistent naming scheme that distinguishes the two cases IIRC, but DAG-level and resource-level works IMO. 

To answer your question directly, each action name only gives access to a single action. This is the same for DAG-level permissions and all other permissions as well. So if I have `DAG:example_python_operator.can_read`, that only gives me the ability to read the `example_python_operator` DAG and nothing else.